crates/core/src/backend.rs

```diff
@@ -424,6 +425,17 @@ pub struct ReadSourceEntry<O> {
     pub open: Option<O>,
 }
 
+impl<O> ReadSourceEntry<O> {
+    fn from_path(path: PathBuf, open: Option<O>) -> Self {
+        let node = Node::new_node(
+            path.file_name().unwrap(),
```

Comment:
Can we return result<self> here to not use unwrap or handle the error?

Reply:
Ah, thanks for the note! I actually wanted to fix that later but obviously forget it..
Is now fixed.